### ISSUE TITLE
Does not work for multiple users?

### ISSUE THREAD

**Issue by Schwefelholz** [scotartt/shell_scripts#1]
Hi,

I tried to adapt your scripts to use them for multiple users on one MacBook.

- I created /etc/auto_**userid** - files per user
- mount points are in a subfolder in /Users/**userid**
- there is a separate fix_mounts.sh for each user in /Users/**userid**/bin
- there is a separate org.autonomous.fixmounts.**userid**.plist for each user in /Library/LaunchDaemons
- I'm pretty sure I changed all necessary settings in the scripts and the plist files.

However, once a mount is not for the correct user, there are remounts carried out, but they are not necessarily for the correct user, but for any user the scripts/services are implemented.
Once I reduce the number of setups to one everything seems to be working OK.

Any ideas/suggestions?

Thanks,
Thomas

**Comment by scotartt** [comment]
For users, you might need to use a _LaunchAgent_, not a LaunchDaemon A LaunchDaemon runs systemwide. A LaunchAgent runs for the user when they log in. As I use a laptop, I'm the only user, and that is why I made a single LaunchDaemon.  The LaunchAgent doesn't run as root though - runs as the user. Which will give it permission issues overriding the "root" mounted remote disk (the very problem we are trying to overcome).

What you've written out seems pretty reasonable to me otherwise.

are the mount points the same for each of the users? I would just adapt the script itself - modify the single user to be an array of users and run the bulk of the shell script in a `for user in $users` loop that starts about line 28, something like this:

```
/* omitted */
autofsname=auto_nas

mnt_usrs="smcphee other user anotheruser" # <-- NEW CODE

mnt_pnt=Mounts

for mnt_usr in $mnt_users; # <-- NEW CODE
do # <-- NEW CODE

t_mnt=/Users/${mnt_usr}/${mnt_pnt}
mounts="special media Erato"

all_mounts=`/sbin/mount | /usr/bin/grep $t_mnt | /usr/bin/grep -v "map $autofsname on $t_mnt"`

d=`/bin/date`
echo "fix_mounts [$d] checking $t_mnt for $mounts"
for mt in $mounts;
do
	full_mount=${t_mnt}/${mt}
	if [[ $all_mounts == *"$full_mount"* ]]; 
	then
		mm=`/bin/echo "$all_mounts" | /usr/bin/grep $full_mount` 
		if [[ ! $mm =~ on.$full_mount.*mounted.by.$mnt_usr ]]; then
			echo "fix_mounts [$d] Remounting: $full_mount - because $mnt_usr not mounted $mm"
			/usr/bin/sudo /sbin/umount $full_mount 
			if [ ! $? ]; then /usr/sbin/diskutil unmount force $full_mount; fi
	        /usr/bin/sudo -u $mnt_usr cd $full_mount 
		fi
	else
		echo "fix_mounts [$d] Not mounted: $full_mount - ignoring"
	fi
done
done # <--NEW CODE
```

Bear in mind I have not checked whether this works, so please be careful!

**Comment by Schwefelholz** [comment]
Actually, all users have their own mount point in /Users/**userid**/mount. This is what drives me around as the respective scripts shouldn't care about other users mount point.
They do have their own /etc/auto_**userid**, too, so the adaption you sent wouldn't work for my situation right now.

I'll give it another try and let you know...

**Comment by scotartt** [comment]
have you adapted this line `mounts="special media Erato"` for each user to list each mount point that user wants?

another problem, the line which finds all the mounts:

```
all_mounts=`/sbin/mount | /usr/bin/grep $t_mnt | /usr/bin/grep -v "map $autofsname on $t_mnt"`
```

Have a look on the shell, manually, to see if this is producing the right output.

**Comment by Schwefelholz** [comment]
Yes, I adapted the `mounts=` line, and `all_mounts=` seems to be OK as well.

I must admit it took me some time to really understand the script (and I'm not sure if that's 100% true right now).
After I made the try yesterday (for four users), I cleaned up this morning and made another approach. Right now, I've implemented the solution for two users and it seems to be working as expected. Maybe there was a typo or I missed something anywhere.
I'll leave it like this for a while and extend to all four users eventually.
